vault/identity_store.go

```diff
@@ -180,100 +185,144 @@ func mfaPaths(i *IdentityStore) []*framework.Path {
+						return i.handleMFAMethodWriteCommon(ctx, req, d, methodType)
+					},
 					DisplayAttrs: &framework.DisplayAttributes{
-						OperationVerb:   "configure",
-						OperationSuffix: "totp-method|totp-method",
+						OperationVerb: "update",
 					},
-					Summary: "Update or create a configuration for the given MFA method",
+					Summary: "Update the configuration for the given MFA method",
 				},
 				logical.DeleteOperation: &framework.PathOperation{
-					Callback: i.handleMFAMethodTOTPDelete,
-					DisplayAttrs: &framework.DisplayAttributes{
-						OperationVerb:   "delete",
-						OperationSuffix: "totp-method|totp-method",
+					Callback: func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+						return i.handleMFAMethodDeleteCommon(ctx, req, d, methodType)
 					},
-					Summary: "Delete a configuration for the given MFA method",
+					Summary: "Delete the given MFA method",
 				},
 			},
 		},
 		{
-			Pattern: "mfa/method/totp/?$",
+			Pattern: "mfa/method/" + methodType + "/?$",
 			DisplayAttrs: &framework.DisplayAttributes{
 				OperationPrefix: "mfa",
-				OperationVerb:   "list",
-				OperationSuffix: "totp-methods",
 			},
+			Fields: methodFields,
 			Operations: map[logical.Operation]framework.OperationHandler{
 				logical.ListOperation: &framework.PathOperation{
-					Callback: i.handleMFAMethodListTOTP,
-					Summary:  "List MFA method configurations for the given MFA method",
+					Callback: func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+						return i.handleMFAMethodListCommon(ctx, req, d, methodType)
+					},
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationSuffix: methodTypeForOpenIDOperationID + "-methods",
+					},
+					Summary: "List MFA method configurations for the given MFA method",
+				},
+				// Conceptually, it would make more sense to treat this as a CreateOperation, but we have to leave it
+				// as an UpdateOperation, because the API was originally released that way, and we don't want to change
+				// the meaning of ACL policies that users have written.
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback: func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+						return i.handleMFAMethodWriteCommon(ctx, req, d, methodType)
+					},
+					DisplayAttrs: &framework.DisplayAttributes{
+						OperationVerb:   "create",
+						OperationSuffix: methodTypeForOpenIDOperationID + "-method",
+					},
+					Summary: "Create the given MFA method",
 				},
 			},
 		},
+	}
+}
+
+func mfaTOTPPaths(i *IdentityStore) []*framework.Path {
+	return makeMFAMethodPaths(
+		mfaMethodTypeTOTP,
+		mfaMethodTypeTOTP,
+		map[string]*framework.FieldSchema{
+			"method_name": {
+				Type:        framework.TypeString,
+				Description: `The unique name identifier for this MFA method.`,
+			},
+			"max_validation_attempts": {
```

Comment:
**Advice for people reviewing this PR:** If you turn on the "Ignore whitespace changes" diff mode, you'll be able to see these long definitions of field schemas are not being changed, only re-indented upon being moved to a different place in the code.